### ISSUE TITLE
Add notebook utility functions

### ISSUE THREAD

**Issue by blink1073** [jupyter/jupyterlab#207]
Need functions to fetch a notebook and start a session for a notebook for standalone applications:

https://gist.github.com/blink1073/ac460d8cc9911d77d82b6d7862f06c55

**Comment by jasongrout** [comment]
Are these convenience functions for standalone notebooks?

**Comment by blink1073** [comment]
For using notebooks (the document, not the ui) in different contexts.

**Comment by ellisonbg** [comment]
What is the status of this one?

**Comment by blink1073** [comment]
Just waiting on someone to implement it.

**Comment by jasongrout** [comment]
Not crucial for an MVP, so moving to backlog.

**Comment by minrk** [comment]
just adding to this that it would also be nice to have one for creating cells without notebooks, as well.

**Comment by ellisonbg** [comment]
And also for rendering outputs on pages...

**Comment by jtpio** [comment]
Looks like this issue can now be closed, since it's possible to build lab-based applications with JupyterLab components.

See the examples: https://github.com/jupyterlab/jupyterlab/tree/main/examples


We discussed this issue at the Jupyter Triage call and in accordance with the rationale the triage tool provided we voted to close this issue as there is an existing extension that addresses this.

<details>
**Rationale**: The requested functionality is already available through existing APIs like `NotebookActions`, `DocumentManager`, `SessionManager`, and `SessionContext`. Adding convenience wrapper functions would increase maintenance burden without significant functional benefit. Developers needing this functionality can already accomplish their goals with existing APIs, and documentation improvements would be more valuable than new utility functions. The low community engagement suggests limited demand for this specific enhancement.
</details>
